gimmemotifs/orthologs.py

```diff
@@ -0,0 +1,502 @@
+    cur.execute("CREATE INDEX idx_name_lower ON genes (gene_name_lower, assembly)")
+    cur.execute("CREATE INDEX idx_id_lower ON genes (gene_id_lower, assembly)")
+    cur.execute("CREATE INDEX idx_orthogroup_assembly ON genes (orthogroup, assembly)")
+    conn.commit()
+
+    return db
+
+
+def make_motif2factors(
+    outfile, new_reference, database_references, motifsandfactors, database
+):
+    """
+    Make a motifs2factors file based on an existing ortolog database.
+
+    We loop over each motif, find which factors belong to it in the old
+    reference, then find all the orthogroups those belong to, and finally assign
+    all the genes of the new reference species that belong to those orthogroups.
+
+    However
+    """
+    factor2orthogroups.cache_clear()
+    conn = sqlite3.connect(database)
+    cur = conn.cursor()
+
+    with open(outfile, "w") as f:
+        f.write(f"Motif\tFactor\tEvidence\tCurated\n")
+        for motif, factors in motifsandfactors.items():
+            # remember if we found any orthologs for all the factors belonging
+            # to our motif
+            motif_set = False
+
+            # get all the orthogroups belonging to all our factors belonging to
+            # our motif. This is not as trivial as it sounds, since gene naming
+            # is a mess
+            orthologousgroups = {
+                item
+                for factor in factors
+                for item in factor2orthogroups(
+                    factor, tuple(database_references), database
+                )
+            }
+            if len(orthologousgroups) > 0:
+                for orthologousgroup in orthologousgroups:
+                    res = cur.execute(
+                        f"SELECT * FROM genes WHERE orthogroup='{orthologousgroup}' AND assembly='{new_reference}'"
+                    ).fetchall()
+
+                    if len(res):
+                        for _, gene_name, _, gene_id, *_ in res:
+                            motif_set = True
+                            if gene_name is not None:
+                                f.write(f"{motif}\t{gene_name}\tOrthologs\tN\n")
+                            else:
+                                f.write(f"{motif}\t{gene_id}\tOrthologs\tN\n")
+            if not motif_set:
+                f.write(f"{motif}\tNO ORTHOLOGS FOUND\tOrthologs\tN\n")
+
+
+@lru_cache(maxsize=99999)
+def factor2orthogroups(factor, references, database):
+    """
+    Returns all the orthogroups a factor belongs to. This is NOT a trivial task.
+
+    Some of the factors shouldn't be considered (blacklisted), some have an
+    ambiguous name that should be renamed, and in general the factor naming is
+    inconsistent.
+
+    We first check whether or not it fits in our ortogroup database (easy!), if
+    not, we start querying mygeneinfo to get more gene aliases to search for in
+    our ortogroups. If that didn't work we query mygeneinfo again, but with a
```

Comment:
Maybe make this configurable? Have 3 documented settings, something like `exact`, `strict` and `lenient`. Although `strict` may not be the best term.